lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+class AwsCompileKafkaEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileKafkaEvents.bind(this),
+    };
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    const arrayOrStringSchema = (pattern) => {
+      return {
+        anyOf: [
+          {
+            type: 'array',
+            minItems: 1,
+            items: {
+              type: 'string',
+              pattern,
+            },
+          },
+          {
+            type: 'string',
+            pattern,
+          },
+        ],
+      };
+    };
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'kafka', {
+      type: 'object',
+      properties: {
+        accessConfigurations: {
+          type: 'object',
+          minProperties: 1,
+          properties: {
+            vpcSubnet: arrayOrStringSchema('subnet-[a-z0-9]+'),
```

Comment:
We don't need to use `arrayOrStringSchema`, as in schema validator we have _to array_ coercion configured to be _on_ (https://ajv.js.org/docs/coercion.html)

So we should simply configure those properties as arrays.  Singular values will automatically be allowed, and in code we can unconditionally expect arrays (as coercion will ensure all values are converted to arrays during validation phase)

Reply:
Thanks for bringing that up @medikoo - I was not aware of that :+1:

Reply:
@medikoo after changing this, should I inline the lambda function or keep as is?  I wasn't sure what was more readable.

Reply:
I would inline the schema (when inspecting the schema, then it's easier to acknowledge what's defined)